Makefile

```diff
@@ -68,23 +68,7 @@ test:
 bench:
 	go test -timeout=30s -bench=. $$(go list ./...)
 
-check: gofmt ineffassign lint staticcheck vet
+check:
+	golangci-lint run -v
```

Comment:
@rolinh how do you feel about having to install `golangci-lint` binary? running this inside docker was too slow, and they don't recommend building from source 🤔

Reply:
Not really fond of it but on the other hand, I don't manage to install it from source because of that same old grpc breaking change issue that breaks the etcd client :confused:  :
```
go: finding module for package github.com/golangci/golangci-lint/cmd/golangci-lint
go: found github.com/golangci/golangci-lint/cmd/golangci-lint in github.com/golangci/golangci-lint v1.40.1
go: finding module for package google.golang.org/grpc/naming
go: finding module for package google.golang.org/grpc/examples/helloworld/helloworld
go: found google.golang.org/grpc/examples/helloworld/helloworld in google.golang.org/grpc/examples v0.0.0-20210519181852-3dd75a6888ce
go: finding module for package google.golang.org/grpc/naming
github.com/cilium/cilium-cli/internal/certs imports
        github.com/cloudflare/cfssl/helpers imports
        github.com/google/certificate-transparency-go imports
        go.etcd.io/etcd imports
        go.etcd.io/etcd/etcdmain imports
        go.etcd.io/etcd/proxy/grpcproxy imports
        google.golang.org/grpc/naming: module google.golang.org/grpc@latest found (v1.38.0), but does not contain package google.golang.org/grpc/naming
```
Another option would be to run it as a docker container, which I think I prefer because this means control over the version that is run.

```
docker run --rm -v $(pwd):/app -w /app docker.io/golangci/golangci-lint:v1.40.1 golangci-lint run -v
```

Reply:
> Another option would be to run it as a docker container, which I think I prefer because this means control over the version that is run.
> 
> ```
> docker run --rm -v $(pwd):/app -w /app docker.io/golangci/golangci-lint:v1.40.1 golangci-lint run -v
> ```

yeah i tried that as well, it's much slower (takes more than 1m each time i run the command)  😐

Reply:
:wave: Switching to golangci-lint would be great, was thinking of proposing the same change.

I see you're using a pre-built golangci-lint GH action here, but on most other CIs it's usually installed as a preparation step as described here: https://golangci-lint.run/usage/install.
```
~ curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.40.1
```

This pins the version of the golangci-lint binary, so whatever runs in CI will always be checked against the same version just to make sure.

Just curious, what's the issue with installing the binary locally? The script above can be used, personally I just install it from AUR, and it's also available on Homebrew for mac users. Never had any trouble with that.

Reply:
i just went with their recommendation but no strong preference so long as we get those nice annotations for lint errors:

> We recommend using our GitHub Action for running golangci-lint in CI for GitHub projects

personally i don't have strong preference regarding installing the binary locally either. one benefit of using docker is that it's easier to control which version everybody uses, but it is super slow on mac 😐

Reply:
updated the pr. in summary:

- still using golangci-lint github actions based on their recommendation.
- `make check` runs golangci-lint in a docker container.
- updated the commit message with an alternative way to run`golangci-lint` with a locally installed binary.

Reply:
> Just curious, what's the issue with installing the binary locally?

First, it requires to install the binary. Second, to keep it updated. Third, if the version doesn't match with the version in the repo, discrepancies along what is being reported will likely emerge. For overall consistency, I think it's best to stick to a single version.

Reply:
1, 2.: AUR/brew have the package, so it's set and forget. Keeping the *repo* version up to date would be more of a chore than a local workstation. :sweat_smile: 
3: Yes, very annoying, but whatever linter errors you get running locally should be indicative anyway. As long as CI has the final word, we're not at risk of breaking the build.

Haven't hit 3. in a while (not since using golangci-lint) since it acts as a buffer. New releases usually appear around Go releases. As long as `.golangci.yaml` is in the repo, my editor will be happy. :shrug: If the Makefile target runs whatever CI runs, it's good as a final check.